tfe/resource_tfe_agent_pool_test.go

```diff
@@ -0,0 +1,157 @@
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFEAgentPoolDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEAgentPool_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEAgentPoolExists(
+						"tfe_agent_pool.foobar", agentPool),
+					testAccCheckTFEAgentPoolAttributes(agentPool),
+					resource.TestCheckResourceAttr(
+						"tfe_agent_pool.foobar", "name", "agent-pool-test"),
+				),
+			},
+
+			{
+				Config: testAccTFEAgentPool_update,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFEAgentPoolExists(
+						"tfe_agent_pool.foobar", agentPool),
+					testAccCheckTFEAgentPoolAttributesUpdated(agentPool),
+					resource.TestCheckResourceAttr(
+						"tfe_agent_pool.foobar", "name", "agent-pool-updated"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckTFEAgentPoolExists(
+	n string, agentPool *tfe.AgentPool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		tfeClient := testAccProvider.Meta().(*tfe.Client)
+
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No instance ID is set")
+		}
+
+		sk, err := tfeClient.AgentPools.Read(ctx, rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		if sk == nil {
+			return fmt.Errorf("agent pool not found")
+		}
+
+		*agentPool = *sk
+
+		return nil
+	}
+}
+
+func testAccCheckTFEAgentPoolAttributes(
+	agentPool *tfe.AgentPool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if agentPool.Name != "agent-pool-test" {
+			return fmt.Errorf("Bad name: %s", agentPool.Name)
+		}
+		return nil
+	}
+}
+
+func testAccCheckTFEAgentPoolAttributesUpdated(
+	agentPool *tfe.AgentPool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if agentPool.Name != "agent-pool-updated" {
+			return fmt.Errorf("Bad name: %s", agentPool.Name)
+		}
+		return nil
+	}
+}
+
+func testAccCheckTFEAgentPoolDestroy(s *terraform.State) error {
+	tfeClient := testAccProvider.Meta().(*tfe.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "tfe_agent_pool" {
+			continue
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No instance ID is set")
+		}
+
+		_, err := tfeClient.AgentPools.Read(ctx, rs.Primary.ID)
+		if err == nil {
+			return fmt.Errorf("agent pool %s still exists", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+const testAccTFEAgentPool_basic = `
+resource "tfe_organization" "foobar" {
+  name  = "tst-terraform"
+  email = "admin@company.com"
+}
+
+resource "tfe_agent_pool" "foobar" {
+  name         = "agent-pool-test"
+  organization = "${tfe_organization.foobar.id}"
+}`
+
+const testAccTFEAgentPool_update = `
+resource "tfe_organization" "foobar" {
+  name  = "tst-terraform"
+  email = "admin@company.com"
+}
+
+resource "tfe_agent_pool" "foobar" {
+  name         = "agent-pool-updated"
+  organization = "${tfe_organization.foobar.id}"
+}`
```

Comment:
it would be good for all of these org names to have a random int on the end like the other resources to avoid tests breaking whenever some of the tests fail

Reply:
Doh this was just forgotten - note (during a rebase) I already converted all the configurations to use functions and not consts, I just forgot to actually add the interpolated value in the string 🤦  Good catch.